projects/swimlane/ngx-ui/src/lib/components/toggle/toggle.component.ts

```diff
@@ -11,45 +21,24 @@ let nextId = 0;
 
 @Component({
   selector: 'ngx-toggle',
-  template: `
-    <div>
-      <input
-        #input
-        class="ngx-toggle-input"
-        type="checkbox"
-        [id]="id"
-        [(ngModel)]="value"
-        [required]="required"
-        [tabIndex]="tabIndex"
-        [disabled]="disabled"
-        [name]="name"
-        (blur)="onBlur()"
-        (change)="onChange()"
-      />
-      <label [attr.for]="id" class="ngx-toggle-label"> </label>
-      <label [attr.for]="id" class="ngx-toggle-text">
-        <span *ngIf="label" [innerHTML]="label"></span>
-        <ng-content></ng-content>
-      </label>
-    </div>
-  `,
+  exportAs: 'ngxToggle',
+  templateUrl: './toggle.component.html',
   styleUrls: ['./toggle.component.scss'],
   encapsulation: ViewEncapsulation.None,
-  providers: [TOGGLE_VALUE_ACCESSOR]
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [TOGGLE_VALUE_ACCESSOR],
+  host: {
+    class: 'ngx-toggle',
+    '[class.disabled]': 'getDisabled'
+  }
 })
 export class ToggleComponent implements ControlValueAccessor {
-  @Input()
-  id: string = `toggle-${++nextId}`;
-  @Input()
-  name: string = null;
-  @Input()
-  disabled: boolean = false;
-  @Input()
-  required: boolean = false;
-  @Input()
-  tabIndex: number = 0;
-  @Input()
-  label: string;
+  @Input() id: string = `toggle-${++nextId}`;
+  @Input() name: string = null;
+  @Input() disabled: boolean = false;
```

Comment:
can we use coerceBoolean and coerceNumber with getters/setters here so we don't have to bind for static inputs?

Reply:
also, has the demo page been set to on-push for this? I didn't see it in the changes.

Reply:
woops, missed the demo page, updated that and added coercion to the inputs